docs/docs/reference/gen_notebooks/feedback_prod.md

```diff
@@ -0,0 +1,179 @@
+
+## Setup
+
+
+```python
+!pip install weave openai streamlit
+```
+
+First, create a file called `secrets.toml` and add an OpenAI key so it works with `[st.secrets](https://docs.streamlit.io/develop/api-reference/connections/st.secrets)`. You can [sign up](https://platform.openai.com/signup) on the OpenAI platform to get your own API key. 
+
+
+```python
+# secrets.toml
+OPENAI_API_KEY = "your OpenAI key"
+```
+
+Next, create a file called `chatbot.py` with the following contents:
+
+
+```python
+# chatbot.py
+
+import weave
+from openai import OpenAI
+import streamlit as st
+
+st.title("Add feedback")
+
+# highlight-next-line
+@weave.op
+def chat_response(prompt):
+    stream = client.chat.completions.create(
+        model="gpt-4o",
+        messages=[
+            {"role": "user", "content": prompt},
+            *[{"role": m["role"], "content": m["content"]} for m in st.session_state.messages]
+        ],
+        stream=True,
+    )
+    response = st.write_stream(stream)
+    return {'response': response}
+
+client = OpenAI(api_key=st.secrets["OPENAI_API_KEY"])
+# highlight-next-line
+weave_client = weave.init('feedback-example')
+
+
+def display_chat_messages():
+    for message in st.session_state.messages:
+        with st.chat_message(message["role"]):
+            st.markdown(message["content"])
+
+def get_and_process_prompt():
+    if prompt := st.chat_input("What is up?"):
+        st.session_state.messages.append({"role": "user", "content": prompt})
+    
+        with st.chat_message("user"):
+            st.markdown(prompt)
+
+        with st.chat_message("assistant"):
+# highlight-next-line            
+            with weave.attributes({'session': st.session_state['session_id']}):
+                result, call = chat_response.call(prompt) # call the function with `.call`, this returns a tuple with a new Call object
+# highlight-next-line
+                st.button(":thumbsup:",   on_click=lambda: call.feedback.add_reaction("👍"), key='up')
+# highlight-next-line
+                st.button(":thumbsdown:", on_click=lambda: call.feedback.add_reaction("👎"), key='down')
+                st.session_state.messages.append({"role": "assistant", "content": result['response']})
+
+def init_chat_history():
+    if "messages" not in st.session_state:
+        st.session_state.messages = st.session_state.messages = []
+
+
+def main():
+    st.session_state['session_id'] = '123abc'
+    init_chat_history()
+    display_chat_messages()
+    get_and_process_prompt()
+
+if __name__ == "__main__":
+    main()
+```
+
+You can run this with `streamlit run chatbot.py`.
+
+Now, you can interact with this application and click the feedback buttons after each response. 
+Visit the Weave UI to see the attached feedback.
+
+## Explanation
+
+If we consider our decorated prediction function as:
+
+We can use it as usal to deliver some model response to the user:
```

Comment:
This line has to be deleted I believe (it belongs to the next code snippet, see below).